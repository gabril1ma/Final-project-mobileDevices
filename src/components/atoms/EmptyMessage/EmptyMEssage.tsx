import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

const EmptyMessage = ({ message }) => {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>{message}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 20,
  },
  text: {
    fontSize: 16,
    color: '#666',
  },
});

export default EmptyMessage;
