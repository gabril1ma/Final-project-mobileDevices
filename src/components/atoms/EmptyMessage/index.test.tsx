import { create, ReactTestRendererJSON } from "react-test-renderer";
import EmptyMessage from "./EmptyMEssage";

export type TTreeRenderer = ReactTestRendererJSON | ReactTestRendererJSON[];

const MOCK_TEXT = "Jest test";
const mockFunction = () => jest.fn();

describe("<EmptyMessage />", () => {
  let tree: TTreeRenderer;
  let testTree: ReactTestRendererJSON;

  beforeEach(() => {
    tree = create(<EmptyMessage message={MOCK_TEXT} />).toJSON();

    testTree = tree[0] ?? tree;
  });

  it("should render component", () => {
    expect(testTree).toMatchSnapshot();
  });

  it("should have one children", () => {
    expect(testTree?.children?.length).toBe(1);
  });
});
