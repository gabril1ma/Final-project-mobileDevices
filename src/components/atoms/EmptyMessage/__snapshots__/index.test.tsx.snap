// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyMessage /> should render component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#666",
        "fontSize": 16,
      }
    }
  >
    Jest test
  </Text>
</View>
`;
